metronome/checkpointing/interpreter/src/io/iohk/metronome/checkpointing/interpreter/InterpreterRPC.scala

```diff
@@ -0,0 +1,35 @@
+package io.iohk.metronome.checkpointing.interpreter
+
+import io.iohk.metronome.checkpointing.models.{Block, Transaction, Ledger}
+import io.iohk.metronome.checkpointing.models.CheckpointCertificate
+
+/** `InterpreterRPC` is the interface that the Service can call on the Interpreter
+  * side to send queries and commands. It provides RPC style methods for some of the
+  * `InterpeterMessage` types, namely the ones `with Request with FromService`.
```

Comment:
"Service" is an ambiguous word. Since this is a very concrete application I think it's both appropriate and much clearer to refer to the `CheckpointingService`.

"on the Interpreter side" - by that I understand this will be called from an interpreter implementation that lives within the PoW node. But that's not the case, right?

Reply:
Hm, maybe you are right. These both live in the `checkpointing` package, so "Interpreter" and "Service" always implicitly mean "Checkpointing Interpreter" and "Checkpointing Service" when they are mentioned in the same sentence and capitalised. I didn't want to concretely mean the `CheckpointingService` class, just the Service and the Interpreter in the ]architecture diagram](https://github.com/input-output-hk/metronome/blob/develop/docs/architecture.png) (where Ranvir just called it the "BFT Service" actually, but that's split into two now).

I'm not the best at naming things though. @dzajkowski was it confusing to you as well?


> "on the Interpreter side" - by that I understand this will be called from an interpreter implementation

Maybe not the best of wording then. By "the Service can call on the Interpreter" I meant that this is an RPC interface representing the Interpreter to the Service.

Reply:
> will be called from an interpreter implementation that lives within the PoW node

As per my [other comment](https://github.com/input-output-hk/metronome/pull/51#discussion_r661281482), the interpreter implementation living inside the PoW node will not call the `InterpeterRPC` but rather be the one that is called - in fact implementing this interface is what it would mean to implement the interpreter. The calling will be done from the Checkpointing Service where a client implementation lives.

Reply:
I would rephrase that sentence to:

```
/** `InterpreterRPC` is the interface that the Checkpointing Service can call to send queries 
  * and commands to the Interpreter.
```

Reply:
Maybe I would also add something like: 

```
* It is also the interface that the Interpreter implements in order to process the said queries 
* and commands. Thus we have separate client- and server-side implementations.
```

It probably sounds obvious to you, but to me it wasn't :)

Reply:
Thanks, added both changes. 

Totally fair that it wasn't obvious. It was inspired by how I did things in discovery, where the RPC interface is implemented both by [DiscoveryNetwork](https://github.com/input-output-hk/scalanet/blob/develop/scalanet/discovery/src/io/iohk/scalanet/discovery/ethereum/v4/DiscoveryNetwork.scala#L31-L40) and the [DiscoveryService](https://github.com/input-output-hk/scalanet/blob/develop/scalanet/discovery/src/io/iohk/scalanet/discovery/ethereum/v4/DiscoveryService.scala#L112), calling each other. Not sure how well it will work in practice, but this is how the dice fell that way in my head.